htmltools/_core.py

```diff
@@ -1122,15 +1123,15 @@ def as_dict(
             href = urllib.parse.quote(s["href"])
             s.update(
                 {
-                    "href": os.path.join(paths["href"], href),
+                    "href": paths["href"] + "/" + href,
```

Comment:
This is a change of behavior in the case that `paths["href"]` has a trailing slash (or `href` has a leading slash, although that probably shouldn't ever happen in practice):

```py
os.path.join('foo/', 'bar.js')
#> 'foo/bar.js'
'foo/' + '/' + 'bar.js'
#> 'foo//bar.js'
```

I think we can use the posixpath module, as in:

```py
import posixpath

posixpath.join(....)
```

The `os.path` docs reference `posixpath` here at the top: https://docs.python.org/3/library/os.path.html

And the `posixpath` [docstring](https://github.com/python/cpython/blob/3.11/Lib/posixpath.py#L9-L10) says the following:
> Some of this can actually be useful on non-Posix systems too, e.g. for manipulation of the pathname component of URLs.

Reply:
Thank you!